src/nvim/syntax.c

```diff
@@ -8251,9 +8256,37 @@ color_name_table_T color_name_table[] = {
   { NULL, 0 },
 };
 
+/// Tranlate terminal color to one of cterm_color_names.
+/// color_to_name can then be used to retrieve RgbValue
+///
+/// @return NULL if could not find
+/// @see color_to_name
+const char *cterm_int2name(int color)
+{
+  int *color_idx = NULL;
+  if (color >= 27 || color < 0) {
+    return NULL;
+  }
+
+  switch (t_colors) {
+    case 8: color_idx = color_numbers_8;break;
+    case 16: color_idx = color_numbers_16;break;
+    case 88: color_idx = color_numbers_88;break;
+    case 256: color_idx = color_numbers_256;break;
+    default: return NULL;
+  };
+
+  for (int i = ARRAY_SIZE(cterm_color_names); i-- >= 0; ) {
+    if (color == color_idx[i]) {
+      return cterm_color_names[i];
+    }
+  }
+
+  return NULL;
+}
```

Comment:
shouldn't this function along with cterm_color_name and color_numbers_X live in one of the terminal libraries ?